ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/SerializationTest.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.tests
+
+import io.ktor.client.features.*
+import io.ktor.client.features.json.*
+import io.ktor.client.request.*
+import io.ktor.client.tests.utils.*
+import io.ktor.http.*
+import kotlinx.serialization.Serializable
+import kotlin.test.*
+
+@Serializable
+class MyCustomObject(val message: String)
+
+class SerializationTest : ClientLoader() {
+
+    @Test
+    fun testSendCustomObject() = clientTests {
+        config {
+            install(JsonFeature)
+        }
+
+        test { client ->
+            assertFailsWith<ClientRequestException> {
```

Comment:
should the exception code (`response.status`) describe the reason of failure (bad content type in header)? If yes -- maybe add the check for it.

Reply:
Nope. The message should be default in status
The custom message is usually used for custom status codes